pkg/controller/infrastructure/infraflow/utils.go

```diff
@@ -164,3 +167,17 @@ func copyMap(src map[string]string) map[string]string {
 	}
 	return dst
 }
+
+func deref[T any](ts []*T) []T {
+	if reflect.TypeOf(ts).Elem().Kind() != reflect.Pointer {
+		panic("dereferenced type is not a pointer")
+	}
+	res := make([]T, 0)
```

Comment:
Very small opinionated remark:
I prefer the simple declaration `var res []T` instead of `make` because it does not allocate memory and res points to nil, while `res := make([]T, 0)` allocates memory and s points to memory to a slice with 0 elements.